pandas/core/groupby.py

```diff
@@ -2368,6 +2368,13 @@ def _get_grouper(obj, key=None, axis=0, level=None, sort=True,
     # axis of the object
     if level is not None:
         if not isinstance(group_axis, MultiIndex):
+            if isinstance(level, list) or isinstance(level, range):
```

Comment:
it there anyware we handle ``range`` as list-like? I think it should be ``list`` or ``tuple`` as the same as ``key``.